packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -959,16 +959,40 @@ export class AttributesController extends ConsumptionBaseController {
         return attributeVersions;
     }
 
+    public async isAPredecessorOf(probedAttribute: LocalAttribute, referencedAttribute: LocalAttribute): Promise<boolean> {
```

Comment:
I find the names `probedAttribute` and `referencedAttribute` hard to unterstand in terms of their order in the succession chain. Additionally, you could get rid of either `isAPredecessor` or `isASuccessor` altogether, as they're doing the same thing in reverse order.

How about you create a single function that does the job and give the arguments well-defined names:

```ts
public async isInSuccession(predecessor: LocalAttribute, successor: LocalAttribute): Promise<boolean> { /* ... */ }
```

Alternative names could be `isSequentialInSuccession` or something alike. Plus, as a final point, if you compare the attribute IDs instead of the attributes you can get rid of one hop of the traversal through the linked list.

Reply:
I vaguely see where this is coming from, but I fail to see how it simplifies things. Maybe let's have a chat about that.

Reply:
Sure, let's have a chat. I'll give you a call later.

Meanwhile, just to summarize:

>I fail to see how it simplifies things

- removing repetitive implementations => less duplication, fewer functions 
- unambiguous parameter names => better ease of use, less errors, a fulfilling life